File: backend/src/main/java/com/interview/platform/mod12_voice_recording/repository/VoiceRecordingRepository.java
package com.interview.platform.mod12_voice_recording.repository;

import com.interview.platform.mod12_voice_recording.model.VoiceRecordingEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

public interface VoiceRecordingRepository extends JpaRepository<VoiceRecordingEntity, UUID> {
}

File: android-app/app/src/main/java/com/interview/platform/domain/repository/VoiceRecordingRepository.kt
package com.interview.platform.domain.repository

import com.interview.platform.domain.model.mod12_voice_recording.VoiceRecording
import com.interview.platform.data.model.Result
import kotlinx.coroutines.flow.Flow

interface VoiceRecordingRepository {
    fun fetchAll(): Flow<Result<List<VoiceRecording>>>
    suspend fun syncData(): Result<Unit>
}
